spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/SecretPropertySourceTests.java

```diff
@@ -0,0 +1,93 @@
+import org.junit.Test;
+
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class SecretPropertySourceTests {
+
+	private static final GcpProjectIdProvider DEFAULT_PROJECT_ID_PROVIDER = () -> "defaultProject";
+
+	@Test
+	public void testNonSecret() {
+		String property = "spring.cloud.datasource";
+		SecretVersionName secretIdentifier =
+				SecretManagerPropertySource.parseFromProperty(property, DEFAULT_PROJECT_ID_PROVIDER);
+
+		assertThat(secretIdentifier).isNull();
+	}
+
+	@Test
+	public void testShortProperty_secretId() {
+		String property = "gcp-secret/the-secret";
+		SecretVersionName secretIdentifier =
+				SecretManagerPropertySource.parseFromProperty(property, DEFAULT_PROJECT_ID_PROVIDER);
+
+		assertThat(secretIdentifier.getProject()).isEqualTo("defaultProject");
+		assertThat(secretIdentifier.getSecret()).isEqualTo("the-secret");
+		assertThat(secretIdentifier.getSecretVersion()).isEqualTo("latest");
+	}
+
+	@Test
+	public void testShortProperty_projectSecretId() {
+		String property = "gcp-secret/the-secret/the-version";
+		SecretVersionName secretIdentifier =
+				SecretManagerPropertySource.parseFromProperty(property, DEFAULT_PROJECT_ID_PROVIDER);
+
+		assertThat(secretIdentifier.getProject()).isEqualTo("defaultProject");
+		assertThat(secretIdentifier.getSecret()).isEqualTo("the-secret");
+		assertThat(secretIdentifier.getSecretVersion()).isEqualTo("the-version");
+	}
+
+	@Test
+	public void testShortProperty_projectSecretIdVersion() {
+		String property = "gcp-secret/my-project/the-secret/2";
+		SecretVersionName secretIdentifier =
+				SecretManagerPropertySource.parseFromProperty(property, DEFAULT_PROJECT_ID_PROVIDER);
+
+		assertThat(secretIdentifier.getProject()).isEqualTo("my-project");
+		assertThat(secretIdentifier.getSecret()).isEqualTo("the-secret");
+		assertThat(secretIdentifier.getSecretVersion()).isEqualTo("2");
+	}
+
+	@Test
+	public void testLongProperty_projectSecret() {
+		String property = "gcp-secret/projects/my-project/secrets/the-secret";
+		SecretVersionName secretIdentifier =
+				SecretManagerPropertySource.parseFromProperty(property, DEFAULT_PROJECT_ID_PROVIDER);
+
+		assertThat(secretIdentifier.getProject()).isEqualTo("my-project");
+		assertThat(secretIdentifier.getSecret()).isEqualTo("the-secret");
+		assertThat(secretIdentifier.getSecretVersion()).isEqualTo("latest");
+	}
+
+	@Test
+	public void testLongProperty_projectSecretVersion() {
+		String property = "gcp-secret/projects/my-project/secrets/the-secret/versions/3";
+		SecretVersionName secretIdentifier =
+				SecretManagerPropertySource.parseFromProperty(property, DEFAULT_PROJECT_ID_PROVIDER);
+
+		assertThat(secretIdentifier.getProject()).isEqualTo("my-project");
+		assertThat(secretIdentifier.getSecret()).isEqualTo("the-secret");
+		assertThat(secretIdentifier.getSecretVersion()).isEqualTo("3");
+	}
```

Comment:
Missing a test for `getSecretPayload`.

Reply:
We'll cover this directly in your configuration tests.

I don't think it's worth making it non-private to write tests for it unlike the `parseProperty` method which had a lot more code.